api/osrd_infra/views/study.py

```diff
@@ -0,0 +1,206 @@
+    search_fields = ("name", "description", "tags")
+    pagination_class = CustomPageNumberPagination
+
+    def get_queryset(self):
+        """this function will disappear to be replaced by the generic search"""
+        queryset = Project.objects.order_by("-pk")
+        name = self.request.query_params.get("name", None)
+        description = self.request.query_params.get("description", None)
+        tags = self.request.query_params.get("tags", None)
+        filters = Q()
+        if name:
+            filters = filters | Q(name__icontains=name)
+        if description:
+            filters = filters | Q(description__icontains=description)
+        if tags:
+            filters = filters | Q(tags__icontains=tags)
+        return queryset.filter(filters)
+
+    @action(url_path="image", detail=True, methods=["get"])
+    def get_image(self, request, pk=None):
+        queryset = Project.objects.all()
+        project = get_object_or_404(queryset, pk=pk)
+        image_db = project.image
+        if image_db:
+            return HttpResponse(image_db, content_type="image/png")
+        return Response({"image": "null"})
+
+    @action(url_path="study_types", detail=True, methods=["get"])
+    def get_types(self, request, pk=None):
+        return Response([x.value for x in StudyType])
+
+    @action(url_path="study_states", detail=True, methods=["get"])
+    def get_states(self, request, pk=None):
+        return Response([x.value for x in StudyState])
+
+    def create(self, request, pk=None):
+        input_serializer = ProjectSerializer(data=request.data, context={"request": request})
+        input_serializer.is_valid(raise_exception=True)
+        if "image" in input_serializer.validated_data.keys():
+            image_db = input_serializer.validated_data["image"]
+            input_serializer.validated_data["image"] = image_db.read()
+            image = Image.open(image_db)
+            response = HttpResponse(content_type="image/png")
+            image.save(response, "PNG")
+        input_serializer.save()
+        return Response(data=input_serializer.data, status=201)
+
+    def partial_update(self, request, pk=None, project_pk=None):
+        project = get_object_or_404(self.get_queryset(), pk=pk)
+        serializer = ProjectSerializer(project, data=request.data, partial=True, context={"request": request})
+        serializer.is_valid(raise_exception=True)
+        print(serializer.validated_data.keys())
+        if "image" in serializer.validated_data.keys():
+            if serializer.validated_data["image"]:
+                image_db = serializer.validated_data["image"]
+                serializer.validated_data["image"] = image_db.read()
+                image = Image.open(image_db)
+                response = HttpResponse(content_type="image/png")
+                image.save(response, "PNG")
+            else:
+                serializer.validated_data["image"] = None
```

Comment:
In this case we should trigger a 4XX error.

Reply:
Why ? It should be possible to delete an image